src/DynamoCore/Models/HomeWorkspaceModel.cs

```diff
@@ -297,6 +323,52 @@ protected virtual void OnEvaluationCompleted(EvaluationCompletedEventArgs e)
             if (handler != null) handler(this, e);
         }
 
+        public override void GetExecutingNodes()
```

Comment:
The name of this method makes me believe I'm going to get a set of nodes returned to me. But this is not the case. Instead, we return void and within the method we modify a bunch of properties on some nodes. I would prefer that this method returned a set of nodes. Then, you could call this method from the `WorkspaceViewModel`, and set the `ShowExecutionPreview` property on all the `NodeViewModels`. 

I agree that this method should continue to live in the model layer, but we should call it and affect changes to the UI from the view model layer and above.